identity/src/peer_id.rs

```diff
@@ -94,6 +95,10 @@ impl PeerId {
     ///
     /// In case the given [`Multiaddr`] ends with `/p2p/<peer-id>`, this function
     /// will return the encapsulated [`PeerId`], otherwise it will return `None`.
+    #[deprecated(
+        since = "0.1.0",
+        note = "`Multiaddr` will become type-safe in a future release, making this function obsolete. See https://github.com/multiformats/rust-multiaddr/issues/73."
+    )]
```

Comment:
Does it make sense to have this deprecation warning? At this stage, there isn't anything users can do about it because https://github.com/multiformats/rust-multiaddr/issues/73 isn't implemented yet. But at the same time, I want to tell users that this will go away. Thoughts?